e_os.h

```diff
@@ -544,6 +544,8 @@ struct servent *getservbyname(const char *name, const char *proto);
 # define CRYPTO_memcmp memcmp
 #endif
 
+#define IF_FAIL_ASSERT(x)     assert(x); if (!(x))
```

Comment:
I'm pondering, give me some time... 
I'm thinking that a surface macro like this isn't the optimal way. I would prefer something that can be used as an expression, like this:

``` C
    if (!alt_assert(expr)) {
        /* signal badness */
   } 
```

What I would like to see is this:

-   when `expr` is false:
    -    alt_assert returns false when `NDEBUG` is defined
    -   alt_assert will bomb when `NDEBUG` is undefined
-   when `expr` is true, alt_assert returns true.